staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go

```diff
@@ -294,14 +294,24 @@ func TestApfExecuteMultipleRequests(t *testing.T) {
 	defer cancel()
 	StartPriorityAndFairnessWatermarkMaintenance(ctx.Done())
 
+	var wg sync.WaitGroup
+	var mu sync.Mutex
+	var err error
 	for i := 0; i < concurrentRequests; i++ {
-		var err error
+		wg.Add(1)
 		go func() {
-			err = expectHTTPGet(fmt.Sprintf("%s/api/v1/namespaces/default", server.URL), http.StatusOK)
+			errLocal := expectHTTPGet(fmt.Sprintf("%s/api/v1/namespaces/default", server.URL), http.StatusOK)
+			if errLocal != nil {
```

Comment:
Is `t` not threadsafe?

```go
if err != nil{
  t.Error(err)
}
```


Seems much simpler if it is.

Reply:
Indeed, much simpler! I can't believe I missed that :D I confused Error (which is concurrency safe) with Fatal, etc. (which isn't concurrency-safe):

> A test ends when its Test function returns or calls any of the methods FailNow, Fatal, Fatalf, SkipNow, Skip, or Skipf. Those methods, as well as the Parallel method, must be called only from the goroutine running the Test function.

>The other reporting methods, such as the variations of Log and Error, may be called simultaneously from multiple goroutines.

https://golang.org/pkg/testing/#T

Done.